src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -1292,6 +1292,24 @@ GUI_TEST_CLASS_DEFINITION(test_7415_3) {
     CHECK_SET_ERR(window1Sequence != window2Sequence, "Sequences are equal");
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7419) {
+    // Copy "samples/Sanger/alignment.ugenedb" to sandbox
+    GTFile::copy(os, dataDir + "samples/Sanger/alignment.ugenedb", sandBoxDir + "test_7419.ugenedb");
+
+    // Open the copied file
+    GTFileDialog::openFile(os, sandBoxDir + "test_7419.ugenedb");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // Now remove it from the disk manually
+    QFile::remove(sandBoxDir + "test_7419.ugenedb");
+
+    // Expected: the message box about lost database -> click OK -> view is closed.
+    GTUtilsDialog::waitForDialog(os, new MessageBoxDialogFiller(os, QMessageBox::Ok, "The document 'test_7419.ugenedb' was removed from its original folder. Therefore, it will be deleted from the current project"));
+
+    // The document update happens each 3 seconds, the messagebox will appear on the closest update
+    GTGlobals::sleep(3000);
```

Comment:
Sleeps are bad for many reasons: they do not guarantee anything and make tests slow.
Wait here until all pending dialog fillers are completed: use "checkNoActiveWaiters"

Reply:
OK. By the way, if you don't want us to use `GTGlobasl::sleep`, make them private

Reply:
This is the goal. Every week I remove dozen of sleeps but there are still a lot of them

The only case when we really need them is when we want 1) to manage some OS-level behavior. For example a sleep() between clicks to avoid double click or 2) in utility classes: like wait up to N seconds until task is finished and fail otherwise. All such utility classes are highly optimized to avoid unnecessary sleeps: checks for results every 100ms